### PR TITLE
Implement transaction storage to include date

### PR DESCRIPTION
**Pull request by ElginL**

### CODE REVIEW

File: src/test/java/seedu/address/model/transaction/DateTest.java [hunk 1]
```diff
@@ -39,9 +39,11 @@ public void isValidDate() {
         assertFalse(Date.isValidDate("39/10/2000")); // invalid day
         assertFalse(Date.isValidDate("-11/11/2000")); // negative integers
         assertFalse(Date.isValidDate("11.223/09/2000.0")); // non-integer values
+        assertFalse(Date.isValidDate("31/02/2000")); // Invalid day in february
```
Comment (Jo3LW) [inline src/test/java/seedu/address/model/transaction/DateTest.java:44]: Great coverage! Maybe adding another test case where the input is the 31st in a month where there are only 30 days would increase coverage. E.g 31/11/2022 Cheers !

**Review by Jo3LW — APPROVED** [review]
LGTM! Great work

File: src/main/java/seedu/address/logic/parser/BuyCommandParser.java [hunk 2]
```diff
@@ -57,18 +54,18 @@ public BuyCommand parse(String args) throws ParseException {
...
-            date = ParserUtil.parseDate(argMultimap.getValue(PREFIX_DATE).orElse(""));
-        }
 
-        Transaction transaction = new BuyTransaction(goods, price, quantity, date);
+            Transaction newTransaction = new BuyTransaction(goods, price, quantity, new Date(output));
+            return new BuyCommand(index, newTransaction);
+        }
 
-        return new BuyCommand(index, transaction);
+        Date date = ParserUtil.parseDate(argMultimap.getValue(PREFIX_DATE).orElse(""));
```
Comment (jicsontoh) [inline src/main/java/seedu/address/logic/parser/BuyCommandParser.java:68]: Good rearrangement of code! Easier to read

**Review by jicsontoh — APPROVED** [review]
LGTM!

**Pull request merged**